macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/PlantUMLGenerator.java

```diff
@@ -33,6 +33,19 @@
 @Role
 public interface PlantUMLGenerator
 {
+    /**
+     * Generate the image in the passed output parameter, using PlantUML.
+     *
+     * @param input     the textual definition input
+     * @param output    the stream into which the generated image will be written to
+     * @param serverURL the optional plantUML server URL (e.g. {@code http://www.plantuml.com/plantuml}. If not an
+     *                  empty string or not null then the URL is called to get the generated image. Otherwise PlantUML works in
+     *                  embedded mode (requires installation of Graphviz locally for most diagram types, and the {@code
+     *                  GRAPHVIZ_DOT} environment variable must be set to point to the path of the GraphViz executable).
+     * @throws IOException when there's a generation or writing error
+     */
+    void outputImage(String input, OutputStream output, String serverURL) throws IOException;
```

Comment:
This will break backward compatibility. Revapi should fail. Also note that the javadoc formatting is not following our best practices but that should be easy to fix. See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/

To make this not break backward compatibiilty, just make it a default method. BTW the javadoc is not correct as it doesn't explain what's the difference with `void outputImage(String input, OutputStream output, String serverURL, String format) throws IOException;` . What format will be used? (this is what needs to be documented).

Are you sure we need this when we can already specify the format? Why is it needed? Thx